### PR TITLE
New packages and imports tour

### PR DESCRIPTION
**Pull request by travissarles**

### CODE REVIEW

**Comment by tpolecat** [comment]
This file needs a `.md` suffix.

File: tutorials/tour/2017-02-25-packages-and-imports [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+```
+This could correspond to the following directory structure: `SelfDrivingCar/src/main/scala/com/google/selfdrivingcar/camera/Lens.scala`.
+
+## Imports
+`import` clauses are for accessing members (classes, traits, functions, etc.) in other packages. An `import` clause is not required for accessing members of the same package. Import clauses are selective:
+```
+import users._  // import everything from the users package
+import users.User  // import the class User
+import users.{User, UserPreferences}  // Only imports selected members
+import users.{UserPreferences => UPrefs}  // import and rename for convenience
```
Comment (tpolecat) [inline tutorials/tour/2017-02-25-packages-and-imports]: You should also mention import hiding, like `import foo.bar. { Qux => _, _ }` which imports everything *except* `Qux`.

File: tutorials/tour/2017-02-25-packages-and-imports [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+```
+In the event there is a naming conflict and you need to import something from the root of the project, prefix the package name with `__root__`:
+```
+package accounts
+
+import __root__.users._
+```
+
+
+Note: The `scala` and `java.lang` packages as well as `object Predef` are imported by default.
```
Comment (tpolecat) [inline tutorials/tour/2017-02-25-packages-and-imports]: This deserves more detail and should be more explicit. What exactly are the default imports? How do explicit imports and import hiding interact with this? We have not mentioned importing symbols from things other than packages, so `import scala.Predef._` will not make sense at this point.

File: tutorials/tour/2017-02-25-packages-and-imports [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+## Creating a package
+Packages are created by declaring one or more package names at the top of a Scala file.
+
+```
+package users
+
+class User
+```
+One convention is to name the package the same as the directory containing the Scala file. The directory structure of an sbt project for `package users` might look like this:
```
Comment (tpolecat) [inline tutorials/tour/2017-02-25-packages-and-imports]: You should probably mention that scala doesn't care about file layout at all. This is just a convention. However some tools might care (I think IntelliJ does).

File: tutorials/tour/2017-02-25-packages-and-imports [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+## Imports
+`import` clauses are for accessing members (classes, traits, functions, etc.) in other packages. An `import` clause is not required for accessing members of the same package. Import clauses are selective:
+```
+import users._  // import everything from the users package
+import users.User  // import the class User
+import users.{User, UserPreferences}  // Only imports selected members
+import users.{UserPreferences => UPrefs}  // import and rename for convenience
+```
+
+One way in which Scala is different from Java is that imports can be used anywhere:
```
Comment (MasseGuillaume) [inline tutorials/tour/2017-02-25-packages-and-imports]: missing:

1. relative imports

(given: foo.bar.buz)

```scala
import foo._
import bar.buz._
```

or

```scala
import foo._, bar.buz._
```

2. package object

**Comment by travis032654** [comment]
@heathermiller ready to merge

**Comment by SethTisue** [comment]
I think this probably still has some further evolution ahead of it, but it seems mergeable to me as it stands. Merging!

**Comment by SethTisue** [comment]
oh, this needs fixing up for the new directory structure...

**Comment by SethTisue** [comment]
cee4a666a6f9c444ab01f7de7b9b334f8af172a7 moves it into place

**Comment by SethTisue** [comment]
argh, now that we have html-proofer running, it shows:

```
- ./_site/tour/packages-and-imports.html
219s
738
  *  internally linking to /ba/tour/packages-and-imports.html, which does not exist (line 594)
219s
739
     <a href="/ba/tour/packages-and-imports.html" class="lang">Bosanski</a>
219s
740
  *  internally linking to /es/tour/packages-and-imports.html, which does not exist (line 598)
219s
741
     <a href="/es/tour/packages-and-imports.html" class="lang">Español</a>
219s
742
  *  internally linking to /ko/tour/packages-and-imports.html, which does not exist (line 602)
219s
743
     <a href="/ko/tour/packages-and-imports.html" class="lang">한국어</a>
219s
744
  *  internally linking to /pl/tour/packages-and-imports.html, which does not exist (line 610)
219s
745
     <a href="/pl/tour/packages-and-imports.html" class="lang">Polski</a>
219s
746
  *  internally linking to /pt-br/tour/packages-and-imports.html, which does not exist (line 606)
219s
747
     <a href="/pt-br/tour/packages-and-imports.html" class="lang">Português (Brasil)</a>
219s
748
htmlproofer 3.7.5 | Error:  HTML-Proofer found 5 failures!
```

**Comment by SethTisue** [comment]
I'll fix it in #984

**Pull request merged**